<?xml version="1.0" ?>
<launch>

  <arg name="mode"       default="sim"/>
  <arg name="debug_mode" default="True"/> <!-- bool -->
  <arg name="filter" default="pf"/>

  <!-- Define args -->
  <arg name="particle_count"                    default="30"/>
  <!-- [x, y, z, roll, pitch, yaw] -->
  <arg name="init_covariance" default="[10, 10, 0.0, 0.0, 0.0, 0.01]"/>
  <arg name="motion_covariance" default="[0, 0, 0.0, 0.0, 0.0, 0.0005]"/>
  <arg name="resampling_noise_covariance" default="[2, 2, 0.0, 0.0, 0.0, 0.0]"/>
  <arg name="measurement_covariance"  default="10"/>

  <arg name="map_frame"                         default="map"/>
  <arg name="odom_frame"                         default="odom"/>
  <arg name="odometry_topic"                    default="/$(arg mode)/odom"/>
  <arg name="mbes_pings_topic"                  default="/$(arg mode)/mbes_pings"/>
  <arg name="average_pose_topic"                default="/$(arg filter)/avg_pose"/>
  <arg name="particle_poses_topic"              default="/$(arg filter)/particle_poses"/>
  <arg name="particle_sim_mbes_topic"           default="/$(arg filter)/sim_mbes"/>
  <arg name="average_mbes_topic"                default="/$(arg filter)/avg_mbes"/>

  <!-- Include rviz -->
<!--   <node type="rviz" name="rviz" pkg="rviz" args="-d $(find bathy_mapper)/rviz/bathy_map.rviz" output="screen" /> -->

  <group unless="$(arg debug_mode)"> <!-- quiet_mode -->
    <node type="auv_pf.py" pkg="auv_particle_filter" name="auv_pf" output="log">
      <param name="particle_count"          type= "int"     value="$(arg particle_count)" />
      <param name="init_covariance"                  value="$(arg init_covariance)" />
      <param name="resampling_noise_covariance"                  value="$(arg resampling_noise_covariance)" />
      <param name="measurement_covariance"                  value="$(arg measurement_covariance)" />
      <param name="motion_covariance"                       value="$(arg motion_covariance)" />
      <param name="map_frame"                               value="$(arg map_frame)" />
      <param name="odom_frame"                               value="$(arg odom_frame)" />

      <param name="odometry_topic"                          value="$(arg odometry_topic)" />
      <param name="mbes_pings_topic"                        value="$(arg mbes_pings_topic)" />
      <param name="average_pose_topic"                      value="$(arg average_pose_topic)" />
	  <param name="average_mbes_topic"                      value="$(arg average_mbes_topic)" />
      <param name="particle_poses_topic"                    value="$(arg particle_poses_topic)" />
      <param name="particle_sim_mbes_topic"                 value="$(arg particle_sim_mbes_topic)" />
      <param name="num_beams_sim" value="10" />
	  <param name="num_beams_real" value="250" />
    </node>
  </group>

  <group if="$(arg debug_mode)"> <!-- display loginfo in terminal -->
    <node type="auv_pf.py" pkg="auv_particle_filter" name="auv_pf" output="screen">
      <param name="particle_count"          type= "int"     value="$(arg particle_count)" />
      <param name="init_covariance"                  value="$(arg init_covariance)" />
      <param name="resampling_noise_covariance"                  value="$(arg resampling_noise_covariance)" />
      <param name="measurement_covariance"                  value="$(arg measurement_covariance)" />
      <param name="motion_covariance"                       value="$(arg motion_covariance)" />
      <param name="map_frame"                               value="$(arg map_frame)" />
      <param name="odom_frame"                               value="$(arg odom_frame)" />

      <param name="odometry_topic"                          value="$(arg odometry_topic)" />
      <param name="mbes_pings_topic"                        value="$(arg mbes_pings_topic)" />
      <param name="average_pose_topic"                      value="$(arg average_pose_topic)" />
  	  <param name="average_mbes_topic"                      value="$(arg average_mbes_topic)" />
      <param name="particle_poses_topic"                    value="$(arg particle_poses_topic)" />
      <param name="particle_sim_mbes_topic"                 value="$(arg particle_sim_mbes_topic)" />
      <param name="num_beams_sim" value="20" />
	  <param name="num_beams_real" value="250" />				

    </node>
  </group>

</launch>

